Lib/Exception.cpp

```diff
@@ -64,7 +64,7 @@ void InvalidOperationException::cry (ostream& str)
 } // InvalidOperationException::cry
 
 
-SystemFailException::SystemFailException(const vstring msg, int err)
+SystemFailException::SystemFailException(const vstring& msg, int err)
```

Comment:
Please note that `msg` might be passed by-value without a reference to constant, because the `vstring` has class has its own allocator (see explanation in the summary).